src/lib/storage/BasicMessageRecord.ts

```diff
@@ -17,7 +18,7 @@ export class BasicMessageRecord extends BaseRecord implements BasicMessageStorag
   public readonly type = BasicMessageRecord.type;
 
   public constructor(props: BasicMessageStorageProps) {
-    super(props.id || uuid());
+    super(props.id || uuid(), props.createdAt || Date.now());
```

Comment:
TypeScript 4.0 introduced some new [short circuit assignment operators](https://devblogs.microsoft.com/typescript/announcing-typescript-4-0/#short-circuiting-assignment-operators). Which are perfect for this usage.

```suggestion
    super(props.id ?? uuid(), props.createdAt ?? Date.now());
```

Reply:
Ah, I see. That's nice. I can't even count how often I solve issues with `if (number) ...` 😄

Reply:
Hmm, as I'm thinking about it. Are they really perfect for this usage? Maybe values `''` and `0` are not quite valid as `id` or `createAt`

Reply:
But that's perhaps the responsibility of some eventual validation. Ok, I'll change it to `??`

Reply:
Well yeah, 0 is probably not a commonly used value. However it is unexpected behaviour IMO to change the value without notic. When I pass a value of `0` and it is automatically changed to `Date.now` without warning or error that is weird I think. Same with empty string.